### ISSUE TITLE
Gutenberg mobile editor: Misaligned Header items

### ISSUE THREAD

**Issue by StefanNieuwenhuis** [Automattic/wp-calypso#50758]
The Header in the Gutenberg Editor on Mobile Web doesn't contain sufficient screen estate to align the items nicely.

<img width="488" alt="Screenshot 2021-03-04 at 14 17 34" src="https://user-images.githubusercontent.com/12104589/109970816-cd696400-7cf5-11eb-8174-c4e04ef1758b.png">

Steps to reproduce

1. Switch to a mobile viewport (I used Moto G4). Here's [a guide](https://developers.google.com/web/tools/chrome-devtools/device-mode#viewport) on how to do it w/ Chrome Dev Tools.
2. Go to `https://wordpress.com/home/<somesiteid>.wordpress.com`
3. From the main menu, select Appearance > Themes
4. Activate a different theme and proceed to edit the homepage
5. The items in the header aren't aligned.

Here's a video where the behavior is being reproduced https://www.loom.com/share/6f6eecb26818454891daaaf3753ca3b6.

You can also go straight to `https://wordpress.com/page/<somesiteid>.wordpress.com/home` to reproduce the behavior.

### Notes
Related to #48337 and #47386. See https://github.com/Automattic/wp-calypso/issues/47386#issuecomment-746800009

**Hint**: to prevent the DOM manipulation that adds the Launch button and modifies the Save button, we can do an early return, on smaller screens: https://github.com/Automattic/wp-calypso/blob/trunk/apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-button/index.ts#L24

**Comment by deBhal** [comment]
This looks to be about having the cutoffs wrong/too small for the adaptive elements in our UI, leading to the elements wrapping around at screen sizes close to the transition points.

![Edit_Post_‹_Test_—_WordPress_com](https://user-images.githubusercontent.com/5952255/110772417-50b32880-82a7-11eb-9c56-ed3a9e00b097.jpg)

![Edit_Post_‹_Test_—_WordPress_com](https://user-images.githubusercontent.com/5952255/110775973-44c96580-82ab-11eb-97d3-54af34ef7f15.jpg)

The problem is exacerbated by more/longer elements in the bar, for example while auto-saving for a jetpack site, the bar will wrap even at 820 pixel screen-width: 

![Edit_Post_‹_Test_—_WordPress_com](https://user-images.githubusercontent.com/5952255/110773742-c966b480-82a8-11eb-9003-45fba50d158d.jpg)


So there are a couple of different fixes to consider:
- decrease the threshold for reducing elements. This is the most obvious fix for the  782px and 600px thresholds, but won't actually address the Moto G4, as that is already using the smallest set of elements.
- add new even smaller UI option, reducing the footprint of one or more of elements in the smallest `@media` case or reducing the minimum whitespace between elements or switching to a more vertical style.
- stop the bar from wrapping (i.e. change the failure state).

I'm leaning towards making the edit-post-header more responsive, preventing wrapping and allowing more whitespace to be removed as it shrinks.

**Comment by simison** [comment]
I think for now we should just hide the button on mobile. It's still available in the customer home page.

@razvanpapadopol @ollierozdarz thoughts? What would be a good alternative placement for this in mobile screens, if any?

**Comment by deBhal** [comment]
@simison : Which button do you mean? The autosaving? That does seem a good target to improve things, but we'll need a lot more.

It's even worse in other languages - check out german, which still doesn't have enough space at 1010px width:
![Seite_bearbeiten_‹_Test_—_WordPress_com](https://user-images.githubusercontent.com/5952255/110874400-66643480-831f-11eb-9264-3128ce74269a.jpg)

Contrast that with Chinese, which could show every element at half that size:
![編輯頁面_‹_Test_—_WordPress_com](https://user-images.githubusercontent.com/5952255/110874619-cd81e900-831f-11eb-9c2f-678d5f09293a.jpg)

**Comment by ollierozdarz** [comment]
> I think for now we should just hide the button on mobile. It's still available in the customer home page.

@simison are you talking about the `Launch` button on mobile? That seems like only element there that could possibly be removed/hidden without affecting the core functionality of the toolbar (saving/previewing/publishing) - although I'd hate to see launch rates decline because of it. 

The difficult thing is that I believe we stack plugins in the toolbar (i.e. Jetpack), so this will become an issue as more plugins are installed regardless? 

Have we ever tested using icons for things like 'preview' on mobile? We could always introduce another 'bar', but that would be three bars stacked and take up a lot of the real estate.

**Comment by simison** [comment]
@ollierozdarz yeah, I meant _Launch_ button on mobile can be hidden because it's still better than the broken header. If there's an alternative that we can implement after hiding, great! :-)

> The difficult thing is that I believe we stack plugins in the toolbar (i.e. Jetpack), so this will become an issue as more plugins are installed regardless?

There was a core issue previously (that got since fixed) where plugins would get stacked even when there wasn't space, but they're since getting hidden in the three dots menu.

@alshakero @roo2 let's hide the launch button for now as a fix.

**Comment by creativecoder** [comment]
Note that this is a problem at various screen sizes, not just mobile:

**< 382px**

<img width="389" alt="gutenberg-header-lt-382" src="https://user-images.githubusercontent.com/1699996/112024208-f5343680-8b01-11eb-9bee-859f1d9ffdb5.png">

**600-728px**

<img width="621" alt="gutenberg-header-600-728" src="https://user-images.githubusercontent.com/1699996/112026489-43e2d000-8b04-11eb-8377-2c889916d4e6.png">

**782-824px**

<img width="811" alt="gutenberg-header-782-824" src="https://user-images.githubusercontent.com/1699996/112026554-50672880-8b04-11eb-8e35-1f69e4388cc6.png">

And as pointed out above, it varies by language and can happen an larger screen sizes with added buttons (autosave) and/or plugin menus.

---

[Here's a draft PR](https://github.com/Automattic/wp-calypso/pull/51282) that hides the Launch button for small screens, but it doesn't help with screen widths above 480px.

Selectively hiding/showing the button based on screen width across a large range of sizes seems error prone and not maintainable... I think we're going to need some other (smarter) collapsing behavior, but I'm not sure what, yet.

**Comment by kwight** [comment]
This is a visual problem that, while pretty ugly, doesn't affect a user's ability to use anything here. Dropping this to normal priority.

**Comment by razvanpapadopol** [comment]
Once the [Launch button is removed from the editor header](https://github.com/Automattic/wp-calypso/issues/53109), can we retest this issue in all the cases (page, post, draft, etc)?

Check out a couple examples of a published page:

* **Mobile landscape (640px)**
<img width="645" alt="Screenshot 2021-05-24 at 17 39 13" src="https://user-images.githubusercontent.com/14192054/119364136-4524aa80-bcb7-11eb-9dff-ee01df072455.png">

* **Some viewport 601px wide**
<img width="601" alt="Screenshot 2021-05-24 at 17 40 45" src="https://user-images.githubusercontent.com/14192054/119364126-448c1400-bcb7-11eb-880e-b1ab1767ff60.png">
